portal-frontend/src/app/services/application/application.service.ts

```diff
@@ -10,37 +11,45 @@ import { ApplicationDto, UpdateApplicationDto } from './application.dto';
 export class ApplicationService {
   private serviceUrl = `${environment.apiUrl}/application`;
 
-  constructor(private httpClient: HttpClient) {}
+  constructor(private httpClient: HttpClient, private toastService: ToastService) {}
 
   async create() {
     try {
       return firstValueFrom(this.httpClient.post<{ fileId: string }>(`${this.serviceUrl}`, {}));
     } catch (e) {
-      //Do something
+      this.toastService.showErrorToast('Failed to create Application, please try again later');
```

Comment:
Looks like 'throw e;' is missing in the service.  If an error happens it will not stop the follow-up logic execute, is it intentional?

Reply:
Any callers to this function will get the type Application | undefined and will need to handle the undefined case. Since this is showing an error toast it made more sense not to re-throw. We could alternatively have it not have any try catch and rely on anyone calling it to try/catch and show the error toast.

Reply:
I think it is fine for now, we probably will revisit this once we get the actual use case. As for me, it is more like what happens, for example with the get list endpoint that returns [] on an error and [] if there are no data to return so the result is not clear in that case. 
As for me, ideally, service methods can show generic errors related to their actions, and client that calls them can override those based on the requirements.

Reply:
Yeah its kinda a stuck in between situation, in some cases its nice to have the service catch and show the toast so that we dont need to duplicate it in everyone calling it, but sometimes we need to override it and show a warning or have other behaviour. I think in this case we would need a safe version that returns Application | undefined and an unsafe version that throws the exception and relies on the callee to catch it. Unfortunately I don't think typescript lets you specify a function that will throw an error and you should catch it.